### ISSUE TITLE
Extend JRE enum to support Java 14

### ISSUE THREAD

**Issue by sormuras** [junit-team/junit5#1934]
## Related Issues

- #1656

## Deliverables

Perform the following for Java 14.

- [ ] Extend `JRE` enum with the new `JAVA_N` constant, with `N` being the latest feature version number.
- [ ] Update `EnabledOnJreIntegrationTests`, `DisabledOnJreIntegrationTests`, `EnabledOnJreConditionTests`, and `DisabledOnJreConditionTests` accordingly.
- [ ] Test against JDK `N`.

**Comment by sormuras** [comment]
_in progress..._

**Comment by sbrannen** [comment]
>     * [ ]  Document in Release Notes.

Will be addressed in #1935.

**Comment by sormuras** [comment]
Groovy-
```
> Task :junit-jupiter-engine:compileTestGroovy FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':junit-jupiter-engine:compileTestGroovy'.
> BUG! exception in phase 'semantic analysis' in source unit
 '...\junit-jupiter-engine\src\test\groovy\org\junit\jupiter\api\GroovyAssertEqualsTests.groovy'
Unsupported class file major version 58
```
... and Mockito-based tests...
```
org.junit.jupiter.params.provider.MethodArgumentsProviderTests > providesArgumentsUsingIterator() FAILED
    org.mockito.exceptions.base.MockitoException:
    Cannot call abstract real method on java object!
    Calling real methods is only possible when mocking non abstract method.
```
...failed to compile/run on JDK 14. JDK 11, 12 and 13 are okay.
